### ISSUE TITLE
Unexpected error (BadDeviceId calling)

### ISSUE THREAD

**Issue by im1x** [iXab3r/MicSwitch#45]
Received an unexpected error.

The error mentions the path "C:\BuildAgent\work", but it doesn't exist on my computer.

Today I upgraded to version 1.0.188.0, I didn’t get this error before.

```
2021-06-21 19:10:38,491 [12] ERROR Unhandled application exception() [PoeShared.SharedLog]
NAudio.MmException: BadDeviceId calling waveOutGetDevCaps
   at NAudio.Wave.WaveOut.GetCapabilities(Int32 devNumber)
   at PoeShared.Audio.Services.AudioPlayer.<>c__DisplayClass3_0.<GetDevices>b__0(Int32 x) in C:\BuildAgent\work\61454f6050c7a045\Sources\PoeShared.Native\Audio\Services\AudioPlayer.cs:line 41
   at System.Linq.Enumerable.SelectRangeIterator`1.MoveNext()
   at System.Linq.Enumerable.ConcatIterator`1.MoveNext()
   at PoeShared.Scaffolding.EnumerableExtensions.ToDictionary[T,TKey,TValue](IEnumerable`1 enumerable, Func`2 keyExtractor, Func`2 valueExtractor, Func`2 conflictSolver) in C:\BuildAgent\work\61454f6050c7a045\Sources\PoeShared\Scaffolding\EnumerableExtensions.cs:line 84
   at PoeShared.Audio.ViewModels.WaveOutDeviceSelectorViewModel.HandleDevicesUpdate()
   at PoeShared.Scaffolding.ObservableExtensions.<>c__DisplayClass5_0`1.<SubscribeSafe>b__0(T x) in C:\BuildAgent\work\61454f6050c7a045\Sources\PoeShared\Scaffolding\ObservableExtensions.cs:line 58
```
![MicSwitch_iiTKVNmk5J](https://user-images.githubusercontent.com/1444150/122796859-895f9680-d2c7-11eb-98d7-0cf652fe817b.jpg)

**Comment by im1x** [comment]
It seems this error occurs every time the computer wakes up from sleep mode. My monitor is connected via HDMI and is visible in audio devices, perhaps the program does not correctly process the moment when the monitor is turned on.

**Comment by iXab3r** [comment]
> It seems this error occurs every time the computer wakes up from sleep mode. My monitor is connected via HDMI and is visible in audio devices, perhaps the program does not correctly process the moment when the monitor is turned on.
That sounds quite right. I will try to reproduce that problem locally. 

> "The error mentions the path "C:\BuildAgent\work", but it doesn't exist on my computer." 
This is where it was built :) It's more of a help for developer than for user

**Comment by iXab3r** [comment]
Fixed crash in 1.0.190. 
Current implementation does devices list update in a safer manner with retry functionality built-in to cover possible edge cases
